Update README for dfx release (#156)
dfinity/motoko-base: README.md

[vessel package manager]: https://github.com/kritzcreek/vessel

> Note: The base library version above advances with the Motoko compiler in a dependent way when new primitive operations appear in the `Prim` module, or they change names.  Mismatched versions result in those primitive operations being unrecognized by the compiler (compiler either too old, or too new, depending), rather than a more explicit "version mismatch"-style error.

Reviewer: ```suggestion
Note: The base library version above advances with the Motoko compiler in a dependent way when the internals of the compiler change. Mismatched versions result in those primitive operations being unrecognized by the compiler (compiler either too old, or too new, depending), rather than a more explicit "version mismatch"-style error.
```
We should avoid pointing to `Prim` too directly in documents aimed at developers, to not tempt them to use it directly.
